e2e/suites/actions/copy.test.ts

```diff
@@ -589,144 +288,368 @@ describe('Copy content', () => {
+
+    await page.goToMyLibraries();
+    await dataTable.doubleClickOnRowByName(siteName);
+    await dataTable.doubleClickOnRowByName(destination);
+
+    expect(await dataTable.isItemPresent(file1)).toBe(true, `${file1} not present in destination folder`);
+    expect(await dataTable.isItemPresent(folder1)).toBe(true, `${folder1} not present in destination folder`);
+    await dataTable.doubleClickOnRowByName(folder1);
+    expect(await dataTable.isItemPresent(fileInFolder)).toBe(true, `${fileInFolder} not present in parent folder`);
+  }
+
+  async function copyLockedFile(destination) {
+    await apis.user.nodes.lockFile(file1Id);
+
+    await dataTable.selectItem(file1);
+    await toolbar.clickMoreActionsCopy();
+    await copyDialog.selectLocation('Personal Files');
+    await copyDialog.selectDestination(destination);
+    await copyDialog.clickCopy();
+    const msg = await page.getSnackBarMessage();
+    expect(msg).toContain('Copied 1 item');
+    expect(msg).toContain('Undo');
+
+    await copyDialog.waitForDialogToClose();
+    expect(await dataTable.isItemPresent(file1)).toBe(true, `${file1} not present in source folder`);
+    await page.clickPersonalFilesAndWait();
+    await dataTable.doubleClickOnRowByName(destination);
+    expect(await dataTable.isItemPresent(file1)).toBe(true, `${file1} not present in destination folder`);
+  }
+
+  async function copyFolderThatContainsLockedFile(destination) {
+    await apis.user.nodes.lockFile(fileInFolderId);
+
+    await dataTable.selectItem(folder1);
+    await toolbar.clickMoreActionsCopy();
+    await copyDialog.selectLocation('Personal Files');
+    await copyDialog.selectDestination(destination);
+    await copyDialog.clickCopy();
+    const msg = await page.getSnackBarMessage();
+    expect(msg).toContain('Copied 1 item');
+    expect(msg).toContain('Undo');
+
+    await copyDialog.waitForDialogToClose();
+    expect(await dataTable.isItemPresent(folder1)).toBe(true, `${folder1} not present in source folder`);
+    await page.clickPersonalFilesAndWait();
+    await dataTable.doubleClickOnRowByName(destination);
+    expect(await dataTable.isItemPresent(folder1)).toBe(true, `${folder1} not present in destination folder`);
+    expect(await dataTable.isItemPresent(fileInFolder)).toBe(false, `${fileInFolder} is present in destination`);
+
+    await dataTable.doubleClickOnRowByName(folder1);
+    expect(await dataTable.isItemPresent(fileInFolder)).toBe(true, `${fileInFolder} is not present in parent folder`);
+  }
+
+  async function undoCopyOfFiles(destination) {
+    await dataTable.selectItem(file4);
+    await toolbar.clickMoreActionsCopy();
+    await copyDialog.selectLocation('Personal Files');
+    await copyDialog.selectDestination(destination);
+    await copyDialog.clickCopy();
+    const msg = await page.getSnackBarMessage();
+    expect(msg).toContain('Copied 1 item');
+    expect(msg).toContain('Undo');
+
+    await page.clickSnackBarAction();
```

Comment:
try moving this line before the 2 expects from above. 
I saw some "stale element reference" errors, and suspect this is cause when it actually gets to this line the snackbar is no longer visible